instrumentation/opentelemetry-api-metrics-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/metrics/bridge/ApplicationDoubleHistogram.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.opentelemetryapi.metrics.bridge;
+
+import application.io.opentelemetry.api.common.Attributes;
+import application.io.opentelemetry.api.metrics.BoundDoubleHistogram;
+import application.io.opentelemetry.api.metrics.DoubleHistogram;
+import application.io.opentelemetry.context.Context;
+import io.opentelemetry.javaagent.instrumentation.opentelemetryapi.context.AgentContextStorage;
+import io.opentelemetry.javaagent.instrumentation.opentelemetryapi.trace.Bridging;
+
+final class ApplicationDoubleHistogram implements DoubleHistogram {
+
+  private final io.opentelemetry.api.metrics.DoubleHistogram agentHistogram;
+
+  ApplicationDoubleHistogram(io.opentelemetry.api.metrics.DoubleHistogram agentHistogram) {
+    this.agentHistogram = agentHistogram;
+  }
+
+  @Override
+  public void record(double value) {
+    agentHistogram.record(value);
+  }
+
+  @Override
+  public void record(double value, Attributes applicationAttributes) {
+    agentHistogram.record(value, Bridging.toAgent(applicationAttributes));
+  }
+
+  @Override
+  public void record(double value, Attributes applicationAttributes, Context applicationContext) {
+    agentHistogram.record(
+        value,
+        Bridging.toAgent(applicationAttributes),
+        AgentContextStorage.getAgentContext(applicationContext));
```

Comment:
In a future PR I will clean up this bridging pattern

Reply:
Happy to talk ideas around this.  I'm a bit stale on java modules, but have done my fair share of classloader sharing magic in the past.